src/js/libcs/OpSound.js

```diff
@@ -1,48 +1,457 @@
 //*******************************************************
 // and here are the definitions of the sound operators
 //*******************************************************
-
 var sound = {};
-sound.lines = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+var lines = [0];
+var linesDict = {0: 0};
+var linesCounter = 1;
 
-evaluator.playsin$1 = function(args, modifs) {
+var getAudioContext = function() {
+  var ac = null;
+  if ( !window.AudioContext && !window.webkitAudioContext ) {
+    console.warn('Web Audio API not supported in this browser');
+  } else {
+    ac = new ( window.AudioContext || window.webkitAudioContext )();
+  }
+  return function() {
+    return ac;
+  };
+}();
+
+var audioCtx = null;
 
+
+evaluator.stopsound$0 = function(){
+    audioCtx.close();
+    audioCtx = null;
+};
+
+evaluator.playsin$1 = function(args, modifs) {
+  if(!audioCtx){
+    audioCtx = getAudioContext();
+  }
     function handleModifs() {
         var erg;
+        if (modifs.amp !== undefined) {
+            erg = evaluate(modifs.amp);
+            if (erg.ctype === 'number') {
+                amp = erg.value.real;
+            }
+        }
+        if (modifs.damp !== undefined) {
+            erg = evaluate(modifs.damp);
+            if (erg.ctype === 'number') {
+                damp = erg.value.real;
+            }
+        }
+        if (modifs.duration !== undefined) {
+            erg = evaluate(modifs.duration);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.stop !== undefined) {
+            erg = evaluate(modifs.stop);
+            if (erg.ctype === 'number') {
+                duration = erg.value.real;
+            }
+        }
+        if (modifs.harmonics !== undefined) {
+            erg = evaluate(modifs.harmonics);
+            if (erg.ctype === 'list') {
+                harmonics = erg.value;
+            }
+        }
         if (modifs.line !== undefined) {
-
             erg = evaluate(modifs.line);
             if (erg.ctype === 'number') {
-                linenumber = Math.floor(erg.value.real);
-                if (linenumber < 0) {
-                    linenumber = 0;
-                }
-                if (linenumber > 10) {
-                    linenumber = 10;
-                }
+                var val = erg.value.real;
+            }
+            else if (erg.ctype === 'string'){
+              var val = erg.value;
+            }
+            else {
+                var val = 0;
+            }
+            if (val === 0) {
+              line = 0;
             }
+            else if (!linesDict[val]){
```

Comment:
better use `!linesDict.hasOwnProperty(val)` here, because `linesDict[0]=0`, and therefore always `!linesDict[0]` holds